### ISSUE TITLE
Using external checks results in 'utf-8' codec can't decode byte error

### ISSUE THREAD

**Issue by syphernl** [bridgecrewio/checkov#1797]
**Describe the bug**
We've been using a repository with custom checks (Python-based):

```bash
checkov -d . -o junitxml --external-checks-git https://${CHECKOV_GIT_USER}:${CHECKOV_GIT_PASS}@${CHECKOV_GIT_REPO}//tests/terraform | tee checkov-report.xml
```

This worked fine up until we suddenly got an error:

```
Traceback (most recent call last):
  File "/usr/local/bin/checkov", line 8, in <module>
    exit(run())
  File "/usr/local/lib/python3.9/site-packages/checkov/main.py", line 182, in run
    scan_reports = runner_registry.run(root_folder=root_folder, external_checks_dir=external_checks_dir,
  File "/usr/local/lib/python3.9/site-packages/checkov/common/runners/runner_registry.py", line 55, in run
    scan_report = runner.run(
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 47, in run
    self._load_files(
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 19, in _load_files
    (definitions[f], definitions_raw[f]) = parse(f)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 34, in parse
    (template, template_lines) = load(filename, allow_nulls)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 22, in load
    content = fp.read()
  File "/usr/local/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd2 in position 8: invalid continuation byte
```

The checks itself haven't changed for the past 8 months, it's not really clear to me what causes this.

**To Reproduce**

* Use External checks based on Python (loosely based on the [example in the docs](https://www.checkov.io/3.Custom%20Policies/Python%20Custom%20Policies.html#example).

**Expected behavior**
No error

**Desktop (please complete the following information):**
 - OS: Docker container
 - Checkov Version: container tag `sha256:734351b3355d2e1f7111d591e5393bc4958c5c6c00b9c1d5dcac46c220cbc63e`

**Additional context**
It was working fine with `sha256:5be4f03623dac3e87ad76d573def02f1426d27aee31b664e48e453d090251547`.

**Comment by iwt-marvynzalewski** [comment]
Same issue with yaml custom check but using version 2.0.456 is fine.

```
Process ForkProcess-10:5:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parallelizer/parallel_runner.py", line 27, in func_wrapper
    result = original_func(item)
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 19, in <lambda>
    results = parallel_runner.run_function(lambda f: (f, parse(f)), files_to_load)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 34, in parse
    (template, template_lines) = load(filename, allow_nulls)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 22, in load
    content = fp.read()
  File "/usr/local/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
Process ForkProcess-10:6:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parallelizer/parallel_runner.py", line 27, in func_wrapper
    result = original_func(item)
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 19, in <lambda>
    results = parallel_runner.run_function(lambda f: (f, parse(f)), files_to_load)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 34, in parse
    (template, template_lines) = load(filename, allow_nulls)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 22, in load
    content = fp.read()
  File "/usr/local/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
Process ForkProcess-10:7:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parallelizer/parallel_runner.py", line 27, in func_wrapper
    result = original_func(item)
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 19, in <lambda>
    results = parallel_runner.run_function(lambda f: (f, parse(f)), files_to_load)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 34, in parse
    (template, template_lines) = load(filename, allow_nulls)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parsers/json/__init__.py", line 22, in load
    content = fp.read()
  File "/usr/local/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
Process ForkProcess-10:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parallelizer/parallel_runner.py", line 27, in func_wrapper
    result = original_func(item)
  File "/usr/local/lib/python3.9/site-packages/checkov/common/runners/runner_registry.py", line 58, in <lambda>
    lambda runner: runner.run(root_folder, external_checks_dir=external_checks_dir, files=files,
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 49, in run
    self._load_files(
  File "/usr/local/lib/python3.9/site-packages/checkov/json_doc/runner.py", line 20, in _load_files
    for file, result in results:
  File "/usr/local/lib/python3.9/site-packages/checkov/common/parallelizer/parallel_runner.py", line 41, in _run_function_multiprocess
    yield parent_conn.recv()
  File "/usr/local/lib/python3.9/multiprocessing/connection.py", line 255, in recv
    buf = self._recv_bytes()
  File "/usr/local/lib/python3.9/multiprocessing/connection.py", line 419, in _recv_bytes
    buf = self._recv(4)
  File "/usr/local/lib/python3.9/multiprocessing/connection.py", line 388, in _recv
    raise EOFError
EOFError
```

**Comment by chitrathakkar** [comment]
Getting the same with checkov version 2.0.660

Traceback (most recent call last):
File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
self.run()
File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
self._target(*self._args, **self._kwargs)
File "/home/lmishra/.local/lib/python3.8/site-packages/checkov/common/parallelizer/parallel_runner.py", line 27, in func_wrapper
result = original_func(item)
File "/home/lmishra/.local/lib/python3.8/site-packages/checkov/json_doc/runner.py", line 19, in <lambda>
results = parallel_runner.run_function(lambda f: (f, parse(f)), files_to_load)
File "/home/lmishra/.local/lib/python3.8/site-packages/checkov/common/parsers/json/__init__.py", line 34, in parse
(template, template_lines) = load(filename, allow_nulls)
File "/home/lmishra/.local/lib/python3.8/site-packages/checkov/common/parsers/json/__init__.py", line 22, in load
content = fp.read()
File "/usr/lib/python3.8/codecs.py", line 322, in decode
(result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte

**Comment by nimrodkor** [comment]
Hmmm. I'm thinking this is related to the addition of the JSON checks runner. Looking into it

**Comment by nimrodkor** [comment]
I believe it was resolved in #2122, released 12 days ago @syphernl  @chitrathakkar 
If you're still seeing this - please reopen this issue 🙏
